deps/build.jl

```diff
@@ -1,28 +1,102 @@
 using CBindingGen
+using Libdl
+import Pkg.TOML
 import P4est_jll
 
-# Get include directory from jll package
-const incdir = joinpath(dirname(dirname(P4est_jll.libp4est_path)), "include")
+
+# setup configuration using ideas from MPI.jl
+config_toml = joinpath(first(DEPOT_PATH), "prefs", "P4est.toml")
+mkpath(dirname(config_toml))
+
+if !isfile(config_toml)
+  touch(config_toml)
+end
+
+config = TOML.parsefile(config_toml)
+
+# P4est.toml has 2 keys
+#  library  = "" (default) | library name/path
+#  include  = "" (default) | include name/path
+
+
+# Step 1: Check environment variables and update preferences accordingly
+if haskey(ENV, "JULIA_P4EST_LIBRARY")
```

Comment:
I still think that a single env variable to specify the prefix would be helpful. The most likely use case for changing the library is on a cluster, where the libraries are typically installed to separate locations. Thus I think it would be nice to have this option as well.

Reply:
Added in 7dc05f4